dev/ppc64le/src/ntt_ppc.S

```diff
@@ -0,0 +1,515 @@
+/*
+ * Copyright (c) The mlkem-native project authors
+ * SPDX-License-Identifier: Apache-2.0 OR ISC OR MIT
+ */
+
+#
+# Copyright 2025- IBM Corp.
+#
+#===================================================================================
+# Written by Danny Tsen <dtsen@us.ibm.com>
+#
+
+#include "../../../common.h"
+#if defined(MLK_ARITH_BACKEND_PPC64LE_DEFAULT) && \
+    !defined(MLK_CONFIG_MULTILEVEL_NO_SHARED)
+/* simpasm: header-end */
+
+#include "consts.h"
+
+#define V_QINV  2
+#define V_NMKQ  5
+#define	V_Z0    7
+#define	V_Z1    8
+#define	V_Z2    9
+#define	V_Z3    10
+#define V_ZETA  10
+
+.machine "any"
+.text
+
+.macro SAVE_REGS
+	stdu	1, -352(1)
+	mflr	0
+	std	14, 56(1)
+	std	15, 64(1)
+	std	16, 72(1)
+	std	17, 80(1)
+	std	18, 88(1)
+	std	19, 96(1)
+	std	20, 104(1)
+	std	21, 112(1)
+	li	10, 128
+	li	11, 144
+	li	12, 160
+	li	14, 176
+	li	15, 192
+	li	16, 208
+	stxvx	32+20, 10, 1
+	stxvx	32+21, 11, 1
+	stxvx	32+22, 12, 1
+	stxvx	32+23, 14, 1
+	stxvx	32+24, 15, 1
+	stxvx	32+25, 16, 1
+	li	10, 224
+	li	11, 240
+	li	12, 256
+	li	14, 272
+	li	15, 288
+	li	16, 304
+	stxvx	32+26, 10, 1
+	stxvx	32+27, 11, 1
+	stxvx	32+28, 12, 1
+	stxvx	32+29, 14, 1
+	stxvx	32+30, 15, 1
+	stxvx	32+31, 16, 1
+.endm
+
+.macro RESTORE_REGS
+	li	10, 128
+	li	11, 144
+	li	12, 160
+	li	14, 176
+	li	15, 192
+	li	16, 208
+	lxvx	32+20, 10, 1
+	lxvx	32+21, 11, 1
+	lxvx	32+22, 12, 1
+	lxvx	32+23, 14, 1
+	lxvx	32+24, 15, 1
+	lxvx	32+25, 16, 1
+	li	10, 224
+	li	11, 240
+	li	12, 256
+	li	14, 272
+	li	15, 288
+	li	16, 304
+	lxvx	32+26, 10, 1
+	lxvx	32+27, 11, 1
+	lxvx	32+28, 12, 1
+	lxvx	32+29, 14, 1
+	lxvx	32+30, 15, 1
+	lxvx	32+31, 16, 1
+	ld	14, 56(1)
+	ld	15, 64(1)
+	ld	16, 72(1)
+	ld	17, 80(1)
+	ld	18, 88(1)
+	ld	19, 96(1)
+	ld	20, 104(1)
+	ld	21, 112(1)
+
+	mtlr	0
+	addi    1, 1, 352
+.endm
+
+.macro Load_4Coeffs start next step
+	mr	9, \start
+	add	10, 7, 9	# J + len*2
+	addi	16, 9, \next
+	addi	17, 10, \step
+	addi	18, 16, \next
+	addi	19, 17, \step
+	addi	20, 18, \next
+	addi	21, 19, \step
```

Comment:
Half of these offsets are not used here, but further down in Loda_4AJ, which is confusing. The code should either be moved, or a comment added.